drivers/ieee802154/ieee802154_cc2520.c

```diff
@@ -354,24 +354,24 @@ static inline void set_reset(struct device *dev, u32_t value)
 {
 	struct cc2520_context *cc2520 = dev->driver_data;
 
-	gpio_pin_write(cc2520->gpios[CC2520_GPIO_IDX_RESET].dev,
-		       cc2520->gpios[CC2520_GPIO_IDX_RESET].pin, value);
+	gpio_pin_set(cc2520->gpios[CC2520_GPIO_IDX_RESET].dev,
+		     cc2520->gpios[CC2520_GPIO_IDX_RESET].pin, value);
 }
 
 static inline void set_vreg_en(struct device *dev, u32_t value)
 {
 	struct cc2520_context *cc2520 = dev->driver_data;
 
-	gpio_pin_write(cc2520->gpios[CC2520_GPIO_IDX_VREG_EN].dev,
-		       cc2520->gpios[CC2520_GPIO_IDX_VREG_EN].pin, value);
+	gpio_pin_set(cc2520->gpios[CC2520_GPIO_IDX_VREG_EN].dev,
+		     cc2520->gpios[CC2520_GPIO_IDX_VREG_EN].pin, value);
 }
 
 static inline u32_t get_fifo(struct cc2520_context *cc2520)
 {
 	u32_t pin_value;
 
-	gpio_pin_read(cc2520->gpios[CC2520_GPIO_IDX_FIFO].dev,
-		      cc2520->gpios[CC2520_GPIO_IDX_FIFO].pin, &pin_value);
+	pin_value = gpio_pin_get(cc2520->gpios[CC2520_GPIO_IDX_FIFO].dev,
```

Comment:
Note that `gpio_pin_get` can return a negative value to indicate that the read attempt failed.  We've generally been returning `pin_value > 0` if the API isn't prepared to deal with runtime failures.

Reply:
That's a logic change in the driver. I am not going to change driver logic atm (no test setup etc...). So let's assume (as it was before) that it's always successful.

Reply:
It's not a change in logic, though it does fix a bug.

In the old API any error would have been in the return value which is discarded, and the pin_value would have been returned as an uninitialized value from the stack.

Here if you return `pin_value > 0` you will return zero in error and low, and 1 in active FIFO signal, which is robust and handles the error case that was previously ignored in a well-defined way.